server/commands/src/impls.rs

```diff
@@ -213,3 +213,23 @@ pub fn say(ctx: &mut CommandCtx, message: TextArgument) -> anyhow::Result<()> {
 
     Ok(())
 }
+
+#[command(usage = "me <action>")]
+pub fn me(ctx: &mut CommandCtx, action: TextArgument) -> anyhow::Result<()> {
```

Comment:
You can use pattern matching in the function signatur for `TextArgument` instead of doing `action.0`.

```rust
fn me(ctx: &mut CommandCtx, TextArgument(action): TextArgument) -> String {
    name
}
```

Reply:
Perhaps it would be better to implement `AsRef<str>` for `TextArgument`.

Reply:
Ooh, that's clever, I didn't know you could do that! Thanks